### PR TITLE
Added PDO batch query test

### PR DESCRIPTION
**Pull request by david-puglielli**

### CODE REVIEW

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        case 4:
+        echo "Testing with buffered cursor...\n";
+        $cursor = 'buffered';
+        break;
+    }
+    return $cursor;
+}
+
+$conn = new PDO("sqlsrv:server=$server;database=$databaseName;", $uid, $pwd);
+$conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: You can connect using AE helper method since you use ColumnMeta below

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+$columns = array(new ColumnMeta('int', $colName[0]),
+                 new ColumnMeta('tinyint', $colName[1]),
+                 new ColumnMeta('smallint',$colName[2]),
+                 new ColumnMeta('bigint', $colName[3]),
+                 new ColumnMeta('bit', $colName[4]));
+
+createTable($conn, $tableName, $columns);
+
+$inputs = array();
+
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: Do you need this line?

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+// Insert each row. Need an associative array to use insertRow()
+for ($i = 0; $i < $expectedRows; ++$i) {
+    $inputs = array();
+    for ($j = 0; $j < $expectedResultSets; ++$j) {
+        $inputs[$colName[$j]] = $data[$j][$i];
+    }
+
+    $stmt = insertRow($conn, $tableName, $inputs);
+    unset($stmt);
+}
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: Might want to unset `$inputs` as well (optional)

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        echo "Testing with keyset cursor...\n";
+        $cursor = 'keyset';
+        break;
+        case 4:
+        echo "Testing with buffered cursor...\n";
+        $cursor = 'buffered';
+        break;
+    }
+    return $cursor;
+}
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: Please use a default case for switch statement for a catch all. Also, please use PDO cursor types instead of `$element`.

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+          SELECT c3_smallint FROM $tableName;
+          SELECT c4_bigint FROM $tableName;
+          SELECT c5_bit FROM $tableName;";
+
+// Test the batch query with different cursor types
+for ($i = 0; $i < sizeof($cursors); ++$i) {
+    try {
+        $cursorType = $cursors[$i];
+        $cursor = printCursor($i);
+
```
Comment (yitam) [inline test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: Here in `printCursor` you can pass `$cursorType` directly. The same goes with `checkColumnsAndRows` such that there is no need to use string variable `$cursor`

File: test/functional/pdo_sqlsrv/pdo_batch_query.phpt (hunk 4 above)
Comment (david-puglielli) [reply test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: The cursor array is an array of arrays (because we have to specify `PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE` if `PDO::ATTR_CURSOR` is `PDO::CURSOR_SCROLL`). I could use a nested switch for the case where `PDO::ATTR_CURSOR` is `PDO::CURSOR_SCROLL`?

Comment (yitam) [reply test/functional/pdo_sqlsrv/pdo_batch_query.phpt]: Can you somehow use only `PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE`?

File: test/functional/sqlsrv/sqlsrv_batch_query.phpt [hunk 6]
```diff
@@ -7,7 +7,7 @@ returned otherwise. For information on the expected behaviour of cursors
 with batch queries, see
 https://docs.microsoft.com/en-us/previous-versions/visualstudio/aa266531(v=vs.60)
 --SKIPIF--
-<?php require('skipif_versions_old.inc'); ?>
+<?php require('skipif.inc'); ?>
 --FILE--
```
Comment (yitam) [inline test/functional/sqlsrv/sqlsrv_batch_query.phpt]: Why this change? Should use the AE skipif

Comment (david-puglielli) [reply test/functional/sqlsrv/sqlsrv_batch_query.phpt]: Hmm, not sure what happened there. Will fix.
